velox/exec/HashAggregation.cpp

```diff
@@ -206,6 +208,11 @@ void HashAggregation::addInput(RowVectorPtr input) {
     if (newDistincts_) {
       // Save input to use for output in getOutput().
       input_ = input;
+    } else if (abandonPartialEarly) {
```

Comment:
@xiaoxmeng 
I believe we need to.
I have seen this is being triggered only when abandonPartialEarly is in effect.
If it is not - the bug is not reproducible.

Reply:
Yes, it is triggered by abandonPartialEarly which is the bug. We could add a check that if partialFull_ is true, then it is because abandonPartialEarly set. But we don't need to clear the flag based on if abandonPartialEarly is set. partialFull_ has to be false if newDistincts_ is false.

Reply:
Yeah we can remove the condition and add `VELOX_CHECK(abandonPartialEarly)` here